SAP/luigi
# Implement intent based navigation

**Description**

To test
- Define intentMapping as in the docu and navigate through  linkManager, i.e.: 
  `LuigiClient.linkManager().navigate('#?Intent=Sales-edit?id=100')`
   **The intent identifier matching pattern can be case insensitive: ?Intent= || ?intent= || ?INTENT= || ....**
- The parameters are optional.
- `semanticObject` must be alphanumeric.
- `action` can contain alphanumeric characters and also the underscore sign '_'
- `semanticObject` and  action are `separated` by a hyphen '-' character, which is illegal to use within either of those.
- Should be able to also access the intent through the direct link from `http://localhost:4200/#Sales-settings?` even after page refresh or subsequent URL change. 
- should work with both hash based and path based routing.
- should be able to access the node parameters in the mf after navigation through [getNodeParams](https://docs.luigi-project.io/docs/luigi-client-api/?section=getnodeparams)
- should work with [openAsSplitView](https://docs.luigi-project.io/docs/luigi-client-api/?section=openassplitview) as well
- should be able to work with [feature toggles](https://docs.luigi-project.io/docs/advanced-scenarios) as well.
- unit test and e2e-tests should be green

References for documentation: 
[ Fiori Launch Pad > Configuring Navigation](https://help.sap.com/viewer/cc1c7615ee2f4a699a9272453379006c/7.5.17/en-US/bd8ae3d327ab4541bcce8e7353c046fc.html)
[Fiori Launch Pad > Intent Based Navigation](https://help.sap.com/viewer/a7b390faab1140c087b8926571e942b7/1709.001/en-US/745586a3468640d48b84ffe0e28fe473.html)
[SAP Hana Developer Guide> Intent Based Navigation in App Launcher Tiles ](https://help.sap.com/viewer/52715f71adba4aaeb480d946c742d1f6/2.0.01/en-US/3c6f19fffffa497f9a674fe035a8db87.html)
[Navigation to a Semantic Object](https://sapui5.hana.ondemand.com/#/topic/1d4a0f94bfee48d1b50ca8084a76beec.html)
